Nix: Fix kserver

ttuegel opened the pull request:
The `kserver` that comes with the Nix package is broken, but this pull request would fix it.

ttuegel committed:
Fix interpreter of installed ng

ttuegel committed:
test.nix: Use kserver

ttuegel commented:
The tests fail because of the lines get wrapped differently, so we need to disable line wrapping in the test suite. There is a command-line option for `kompile` and `krun` to do this.

andreiburdusa reviewed (approved):
The only documentation I could find about `buildMaven` is this https://github.com/fzakaria/nixpkgs/blob/b9321adf1ea91bca9e2127ce4f0b428391745474/doc/languages-frameworks/maven.section.md and it doesn't give details about `preFixup`, `doInstallCheck` or `installCheckPhase`. Is there another place that explains all its parameters?

ttuegel commented:
Run tests individually: `nix-build test.nix --argstr test NAME`

ttuegel committed:
test.nix: Find package version file

ttuegel committed:
test.nix: Do not use kserver in Nix

Nailgun seems to want network access which is disabled in the Nix sandbox, so we
can't actually run the tests with kserver enabled.

ttuegel committed:
Merge branch 'master' into nix-ng
